src/test/mir-opt/const_prop/boxes.rs

```diff
@@ -0,0 +1,54 @@
+// compile-flags: -O
+
+#![feature(box_syntax)]
+
+// Note: this test currently doesn't optimize out the box
+// via const prop because box isn't allowed in const eval yet.
```

Comment:
We should never optimize box in const prop. This would end up causing a heap allocation to be in fact a single global unnamed static, which runtime code could want to deallocate later. Or mutate.

Reply:
I updated this comment to say that we can't const prop `box`.